### ISSUE TITLE
[Bug]: ExpectedEOB update 200W records

### ISSUE THREAD

**Issue by zhangxu19830126** [matrixorigin/matrixone#14139]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

latest

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> update news_detail set province = null where province not in ('上海市','云南省','内蒙
古自治区','北京市','台湾省','吉林省','四川省','天津市','宁夏回族自治区','安徽省','山东省','山西省','广东省','广西壮族自治区','新疆维吾尔自治区','江苏省','江西省','河北省','河南省','浙江
省','海南省','湖北省','湖南省','澳门特别行政区','甘肃省','福建省','西藏自治区','贵州省','辽宁省','重庆市','陕西省','青海省','香港特别行政区','黑龙江省') and id between 50001 and 70000;
ERROR: 
ExpectedEOB

### Steps to Reproduce

```Markdown
1. source data
2. update news_detail set province = null where province not in ('上海市','云南省','内蒙古自治区','北京市','台湾省','吉林省','四川省','天津市','宁夏回族自治区','安徽省','山东省','山西省','广东省','广西壮族自治区','新疆维吾尔自治区','江苏省','江西省','河北省','河南省','浙江省','海南省','湖北省','湖南省','澳门特别行政区','甘肃省','福建省','西藏自治区','贵州省','辽宁省','重庆市','陕西省','青海省','香港特别行政区','黑龙江省');
```

**Comment by XuPeng-SH** [comment]
same as #14137

**Comment by aronchanisme** [comment]
```sql
update news_detail set province = null where province not in ('上海市','云南省','内蒙古自治区','北京市','台湾省','吉林省','四川省','天津市','宁夏回族自治区','安徽省','山东省','山西省','广东省','广西壮族自治区','新疆维吾尔自治区','江苏省','江西省','河北省','河南省','浙江省','海南省','湖北省','湖南省','澳门特别行政区','甘肃省','福建省','西藏自治区','贵州省','辽宁省','重庆市','陕西省','青海省','香港特别行政区','黑龙江省');
```

**Comment by aronchanisme** [comment]
2024/1/24:

1. branch=`main`, commit_id=`32a05ab70`, result=`ok`
```bash
mysql> update news_detail set province = null where province not in ('上海市','云南省','内蒙古自治区','北京市','台湾省','吉林省','四川省','天津市','宁夏回族自治区','安徽省','山东省','山西省','广东省','广西壮族自治区','新疆维吾尔自治区','江苏省','江西省','河北省','河南省','浙江省','海南省','湖北省','湖南省','澳门特别行政区','甘肃省','福建省','西藏自治区','贵州省','辽宁省','重庆市','陕西省','青海省','香港特别行政区','黑龙江省');
Query OK, 6218616 rows affected (41.16 sec)

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 32a05ab70     |
+---------------+
1 row in set (0.00 sec)

mysql> select count(*) from news_detail where province is null;
+----------+
| count(*) |
+----------+
|  3109308 |
+----------+
1 row in set (51.36 sec)
```

2. branch=`1.1-dev`, commit_id=`5d7bb0afc`, result=`ok`
```bash
mysql> update news_detail set province = null where province not in ('上海市','云南省','内蒙古自治区','北京市','台湾省','吉林省','四川省','天津市','宁夏回族自治区','安徽省','山东省','山西省','广东省','广西壮族自治区','新疆维吾尔自治区','江苏省','江西省','河北省','河南省','浙江省','海南省','湖北省','湖南省','澳门特别行政区','甘肃省','福建省','西藏自治区','贵州省','辽宁省','重庆市','陕西省','青海省','香港特别行政区','黑龙江省');
Query OK, 3109308 rows affected (52.81 sec)

mysql> select count(*) from news_detail where province is null;
+----------+
| count(*) |
+----------+
|  3109308 |
+----------+
1 row in set (0.23 sec)

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 5d7bb0afc     |
+---------------+
1 row in set (0.00 sec)
```

3. branch=`1.0-dev`, commit_id=``, result=`nok`
```bash
mysql> update news_detail set province = null where province not in ('上海市','云南省','内蒙古自治区','北京市','台湾省','吉林省','四川省','天津市','宁夏回族自治区','安徽省','山东省','山西省','广东省','广西壮族自治区','新疆维吾尔自治区','江苏省','江西省','河北省','河南省','浙江省','海南省','湖北省','湖南省','澳门特别行政区','甘肃省','福建省','西藏自治区','贵州省','辽宁省','重庆市','陕西省','青海省','香港特别行政区','黑龙江省');
ERROR 20101 (HY000): internal error: panic too large fixed slice 2097152, max is 1048576: 
github.com/matrixorigin/matrixone/pkg/lockservice.(*fixedSlicePool).acquire
        /data/mo/1.0-dev/matrixone/pkg/lockservice/slice.go:227
github.com/matrixorigin/matrixone/pkg/lockservice.(*cowSlice).append
        /data/mo/1.0-dev/matrixone/pkg/lockservice/slice.go:71
github.com/matrixorigin/matrixone/pkg/lockservice.(*activeTxn).lockAdded
        /data/mo/1.0-dev/matrixone/pkg/lockservice/txn.go:108
github.com/matrixorigin/matrixone/pkg/lockservice.(*loc
mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 7f65f4ddc     |
+---------------+
1 row in set (0.00 sec)
```
